astropy/visualization/wcsaxes/core.py

```diff
@@ -746,6 +746,44 @@ def tick_params(self, axis='both', **kwargs):
                 if spine in coord.axislabels.get_visible_axes():
                     coord.tick_params(**kwargs)
 
+    def set_world_aspect(self, aspect, **kwargs):
+        """
+        Set the aspect of the axes in world coordinates.
+
+        Parameters
+        ----------
+        aspect : float
+             A circle will be stretched such that the height is *aspect*
+             times the width in world coordinates.
+        kwargs : dict
+            Keyword arguments are handed to
+            :meth:`~matplotib.axes.Axes.set_aspect`.
+
+        Warnings
+        --------
+        Setting an equal world aspect only makes sense when the transformation
+        from pixel to world coordinates is linear.
+
+        Notes
+        -----
+        The aspect is calculated using the ratio of the difference between
+        pixel and world coordinates at (0, 0) and (1, 1) in pixel coordinates.
+        """
+        xunit, yunit = self.wcs.world_axis_units
+        x0, y0 = self.wcs.pixel_to_world_values((0, ), (0, ))
+        x1, y1 = self.wcs.pixel_to_world_values((1, ), (1, ))
```

Comment:
How does this work with WCSes that have more than two world coordinates?

Reply:
This will also give different answers depending on which pixels you use to measure the cdelts